pkg/grpc/jwt_authenticator.go

```diff
@@ -0,0 +1,140 @@
+package grpc
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/buildbarn/bb-storage/pkg/clock"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+// JWTKeyConfig represents an object that can return key content. It is used to bridge
+// configuration to this package.
+type JWTKeyConfig struct {
```

Comment:
This struct doesn't carry its own weight, right? Maybe better to just add a `key interface{}` parameter to `NewJWTAuthenticator()`?